src/main/java/revai/StreamingClient.java

```diff
@@ -0,0 +1,300 @@
+    createWebSocketConnection(url, listener);
+  }
+
+  /**
+   * Overload of {@link StreamingClient#connect(RevAiWebSocketListener, String, StreamContentType,
+   * String, Boolean)} without the optional metadata.
+   */
+  public void connect(
+          RevAiWebSocketListener revAiWebSocketListener,
+          StreamContentType streamContentType,
+          String customVocabularyId,
+          Boolean filterProfanity) {
+    Listener listener = new Listener(revAiWebSocketListener);
+    String url = buildURL(null, streamContentType, customVocabularyId, filterProfanity);
+    createWebSocketConnection(url, listener);
+  }
+
+  /**
+   * Overload of {@link StreamingClient#connect(RevAiWebSocketListener, String, StreamContentType,
+   * String, Boolean)} without the optional customVocabularyId.
+   */
+  public void connect(
+          RevAiWebSocketListener revAiWebSocketListener,
+          String metadata,
+          StreamContentType streamContentType,
+          Boolean filterProfanity) {
+    Listener listener = new Listener(revAiWebSocketListener);
+    String url = buildURL(metadata, streamContentType, null, filterProfanity);
+    createWebSocketConnection(url, listener);
+  }
+
+  /**
+   * Overload of {@link StreamingClient#connect(RevAiWebSocketListener, String, StreamContentType,
+   * String, Boolean)} without the optional metadata and customVocabularyId.
+   */
+  public void connect(
+          RevAiWebSocketListener revAiWebSocketListener,
+          StreamContentType streamContentType,
+          Boolean filterProfanity) {
+    Listener listener = new Listener(revAiWebSocketListener);
+    String url = buildURL(null, streamContentType, null, filterProfanity);
+    createWebSocketConnection(url, listener);
+  }
+
+  /**
+   * Sends data over WebSocket in the form of a ByteString
+   *
+   * @param byteString Audio data in the form of a ByteString
+   */
+  public void sendAudioData(ByteString byteString) {
+    webSocket.send(byteString);
+  }
+
+  /** Will close the WebSocket connection by informing the server that it's the End of Stream */
+  public void close() {
+    webSocket.send("EOS");
+  }
+
+  private void createWebSocketConnection(String url, Listener listener) {
+    Request request = new Request.Builder().url(url).build();
+    webSocket = client.newWebSocket(request, listener);
+  }
+
+  private String buildURL(
+      String metadata,
+      StreamContentType streamContentType,
+      String customVocabularyId,
+      Boolean filterProfanity) {
+
+    HttpUrl.Builder urlBuilder = new HttpUrl.Builder();
+    if (scheme != null) {
+      urlBuilder.scheme(scheme);
+    } else {
+      urlBuilder.scheme("https");
+    }
+    if (host != null) {
+      urlBuilder.host(host);
+    } else {
+      urlBuilder.host("api.rev.ai");
+    }
+    if (port != null) {
+      urlBuilder.port(port);
+    }
+    urlBuilder.addPathSegment("speechtotext");
```

Comment:
super minor, i think you could also do `addPathSegments("speechtotext/v1/stream")` to be a bit less verbose. (note the different method name)